app/controllers/pensions/annualAllowance/ReducedAnnualAllowanceTypeController.scala

```diff
@@ -0,0 +1,97 @@
+                                                     view: ReducedAnnualAllowanceTypeView,
+                                                     clock: Clock) extends FrontendController(mcc) with I18nSupport {
+
+  def show(taxYear: Int): Action[AnyContent] = authAction.async { implicit request =>
+    pensionSessionService.getPensionsSessionDataResult(taxYear, request.user) {
+      case Some(data) =>
+        if (data.pensions.pensionsAnnualAllowances.reducedAnnualAllowanceQuestion.contains(true)) {
+          val taperedAnnualAllowance = data.pensions.pensionsAnnualAllowances.taperedAnnualAllowance
+          val moneyPurchaseAnnualAllowance = data.pensions.pensionsAnnualAllowances.moneyPurchaseAnnualAllowance
+          val form = ReducedAnnualAllowanceTypeQuestionForm.reducedAnnualAllowanceTypeForm(request.user.isAgent)
+          Future.successful(Ok(view(form, taxYear, moneyPurchaseAnnualAllowance, taperedAnnualAllowance)))
+        } else {
+          // navigate to annual allowance CYA when available
+          Future.successful(Redirect(PensionsSummaryController.show(taxYear)))
+        }
+      case _ =>
+        // navigate to annual allowance CYA when available
+        Future.successful(Redirect(PensionsSummaryController.show(taxYear)))
+    }
+
+  }
+
+  def submit(taxYear: Int): Action[AnyContent] = authAction.async { implicit request =>
+    pensionSessionService.getPensionsSessionDataResult(taxYear, request.user) {
+      case Some(data) =>
+        if (data.pensions.pensionsAnnualAllowances.reducedAnnualAllowanceQuestion.contains(true)) {
+          ReducedAnnualAllowanceTypeQuestionForm.reducedAnnualAllowanceTypeForm(request.user.isAgent).bindFromRequest().fold(
+            formWithErrors => Future.successful(BadRequest(view(formWithErrors, taxYear))),
+            reducedAllowanceTypeSelections => {
+
+              val pensionsCYAModel: PensionsCYAModel = data.pensions
+              val pensionsAnnualAllowances = pensionsCYAModel.pensionsAnnualAllowances
+
+              val updatedCyaModel: PensionsCYAModel = {
+                pensionsCYAModel.copy(pensionsAnnualAllowances = pensionsAnnualAllowances.copy(
+                  moneyPurchaseAnnualAllowance = Some(reducedAllowanceTypeSelections.containsMoneyPurchase),
+                  taperedAnnualAllowance = Some(reducedAllowanceTypeSelections.containsTapered)))
+              }
+              pensionSessionService.createOrUpdateSessionData(request.user,
+                updatedCyaModel, taxYear, data.isPriorSubmission)(errorHandler.internalServerError()) {
+                // navigate to 'Have you gone above your reduced annual allowance?' page when available
+                Redirect(PensionsSummaryController.show(taxYear))
+              }
+            }
+          )
+        } else {
+          // navigate to annual allowance CYA when available
+          Future.successful(Redirect(PensionsSummaryController.show(taxYear)))
```

Comment:
Should this one also redirect to ""Have you gone above the annual allowance?" question page" e.g. if it's been answered false or is None

Reply:
yes
But it is the missing question page 'have you gone above the annual allowance page 'rather then the next question" suggested

So changed to:
//TODO: navigate to 'Do you have a reduced annual allowance?' page when available